### PR TITLE
Externalize toggle ui js

### PR DESCRIPTION
**Pull request by orangejenny**
Review commit-by-commit.

@mkangia

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 11 errors:

* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 9 - Missing semicolon. (semi)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 30 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 39 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 53 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 66 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 71 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 73 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 4 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 7 - 'iDataIndex' is defined but never used (no-unused-vars)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 16 - 'HQReportDataTables' is not defined. (no-undef)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 23 - 'value' is defined but never used (no-unused-vars)

**Comment by orangejenny** [comment]
ignore dimagimon

**Comment by snopoke** [comment]
nice

**Comment by dimagimon** [comment]
There are 11 errors:

* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 9 - Missing semicolon. (semi)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 30 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 39 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 53 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 66 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 71 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js, line 73 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 4 - Missing trailing comma. (comma-dangle)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 7 - 'iDataIndex' is defined but never used (no-unused-vars)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 16 - 'HQReportDataTables' is not defined. (no-undef)
* corehq/apps/toggle_ui/static/toggle_ui/js/flags.js, line 23 - 'value' is defined but never used (no-unused-vars)

**Review by mkangia — APPROVED** [review]
Seems mostly code movement. Added minor suggestions if seems valuable.

File: corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                });
+            });
+        };
+
+        self.addItem = function (namespace) {
+            self.items.push({
+                namespace: ko.observable(self.padded_ns[namespace]),
+                value: ko.observable(),
+                last_used: ko.observable()
+            });
```
Comment (orangejenny) [inline corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js]: Not quite following - `self.items.push` does happen twice, but with slightly different parameters.

File: corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                    },
+                    dataType: 'json',
+                    success: function (data) {
+                        self.init_items(data);
+                    }
+                });
+            }
+        });
+
+        var projectInfoUrl = '<a href="' + hqImport('hqwebapp/js/urllib.js').reverse('domain_internal_settings') + '">domain</a>';
```
Comment (orangejenny) [inline corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js]: Agree this isn't great to read, but ultimately the template uses an `html` bind, so it does need a string, not an element. To be fair, that code could be reworked, but I'm trying not to make a whole lot of changes when splitting out js files (same reason that I'm ignoring the linter warnings).

File: corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js (hunk 1 above)
Comment (mkangia) [inline corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js]: Not completely sure how `observable` behaves. Was thinking something like
```
self.pushItem = function(namespace, value, last_used)
```
and then calling self.pushItem(namespace, null_or_some_default_value, {}_or_some_default_value) for the above case.
Can ignore this if thats not feasible.

Comment (orangejenny) [inline corehq/apps/toggle_ui/static/toggle_ui/js/edit-flag.js]: Ah, I see what you're saying, but don't think it's enough cleaner / more concise to be worth it.

**Comment by orangejenny** [comment]
All comments on this are resolved, merging.

**Pull request merged**